consensus/safety-rules/src/safety_rules_2chain.rs

```diff
@@ -90,6 +93,41 @@ impl SafetyRules {
         Ok(vote)
     }
 
+    // TODO: Are these safety rules exhaustive?
+    pub(crate) fn guarded_construct_and_sign_order_vote(
+        &mut self,
+        order_vote_proposal: &OrderVoteProposal,
+    ) -> Result<OrderVote, Error> {
+        // Exit early if we cannot sign
+        self.signer()?;
+        self.verify_order_vote_proposal(order_vote_proposal)?;
+        let proposed_block = order_vote_proposal.block();
+        let mut safety_data = self.persistent_storage.safety_data()?;
+
+        // if already voted on this round, send back the previous vote
+        if let Some(order_vote) = safety_data.last_order_vote.clone() {
```

Comment:
I would prefer resigning order vote instead of keeping it in the safety data.

Reply:
Done.